R/zchunk_LA124.heat.R

```diff
@@ -69,117 +77,121 @@ module_energy_LA124.heat <- function(command, ...) {
+      filter(sector == "out_electricity_heat" , GCAM_region_ID %in% heat_regionIDs$GCAM_region_ID) %>%
       mutate(sector = sub("out_", "", sector)) %>%
-      left_join(enduse_fuel_aggregation, by="fuel")  %>%
-      select(-fuel, -electricity, -bld, -industry, -trn)%>%
-      rename(fuel=heat) %>%
-      gather(key=year, value=value, -fuel, -sector, -GCAM_region_ID) %>%
+      left_join(enduse_fuel_aggregation, by = "fuel") %>%
+      select(GCAM_region_ID, sector, year, value, heat) %>%
+      rename(fuel = heat) %>%
       group_by(fuel, sector, GCAM_region_ID, year) %>%
-      summarise(value=sum(value)) %>%
-      filter(fuel != is.na(fuel)) %>%
-      mutate(year = sub("X", "", year)) %>%
-      mutate(year = as.integer(year)) -> L124.out_EJ_R_heatfromelec_F_Yh
+      summarise(value = sum(value)) %>%
+      filter(fuel != is.na(fuel)) -> L124.out_EJ_R_heatfromelec_F_Yh
 
 
-    #Secondary output coefficients on heat produced by main activity CHP plants
+    # Secondary output coefficients on heat produced by main activity CHP plants
     L1231.out_EJ_R_elec_F_tech_Yh %>%
-      filter(GCAM_region_ID %in% heat_regionIDs) %>%
+      filter(GCAM_region_ID %in% heat_regionIDs$GCAM_region_ID) %>%
+      # Select only technologies that have heat output in calibrated techs mapping
       filter(technology %in% calibrated_techs$technology[calibrated_techs$secondary.output == "heat"]) %>%
-      gather(key=year, value=value, -fuel, -sector, -GCAM_region_ID, -technology) %>%
-      mutate(year = sub("X", "", year)) %>%
-      mutate(year = as.integer(year)) %>%
       left_join(L124.out_EJ_R_heatfromelec_F_Yh %>%
-                  rename(value_elec=value) %>%
-                  rename(temp=sector), by=c("GCAM_region_ID", "fuel", "year")) %>%
-      mutate(value=value_elec/value) %>%
-      select(-value_elec, -temp)-> L124.heatoutratio_R_elec_F_tech_Yh
+                  rename(value_elec = value) %>%
+                  rename(temp = sector), by=c("GCAM_region_ID", "fuel", "year")) %>%
+      mutate(value = value_elec / value) %>%
+      select(GCAM_region_ID, sector, fuel, technology, year, value) %>%
+      # Reset missing and infinite values (applicable for CC in the base years) to 0
+      mutate(value = if_else(is.na(value)|is.infinite(value), 0, value)) -> L124.heatoutratio_R_elec_F_tech_Yh
 
-    #Reset missing and infinite values (applicable for CC in the base years) to 0
-    L124.heatoutratio_R_elec_F_tech_Yh %>%
-      mutate(value=if_else(is.na(value)|is.infinite(value), 0, value))-> L124.heatoutratio_R_elec_F_tech_Yh
 
-    #Drop all rows where all years are 0
+    # Drop all rows where all years are 0
+    # Create a table of all the years of all 0
     L124.heatoutratio_R_elec_F_tech_Yh %>%
       group_by(GCAM_region_ID, technology, sector, fuel) %>%
-      summarise(sum=sum(value)) %>%
-      filter(sum==0) -> all_years_0
+      summarise(sum = sum(value)) %>%
+      filter(sum == 0) -> all_years_0
```

Comment:
maybe something more descriptive than all_years_0 would be helpful. Such as tech_null_outputs or something like that. I'm not even sure whether these rows are.